include/directx/d3dx12_render_pass.h

```diff
@@ -34,6 +34,11 @@ inline bool operator==( const D3D12_RENDER_PASS_BEGINNING_ACCESS_CLEAR_PARAMETER
     return a.ClearValue == b.ClearValue;
 }
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcovered-switch-default"
```

Comment:
I'm curious why you chose to add `default:` and also suppress this. Looks like this warning should only show up if there is a `default` but also all cases are covered, which isn't what's happening here. I would've expected either:
* Add the default to cover the not-handled cases.
* Add the not-handled cases and don't add a default.

Reply:
I'm very confused why `-Wcovered-switch-default` is being generated... The switch definitely doesn't cover all enum values (there's preserve/discard) and that warning should only be generated if the `default:` case is effectively dead.